mgrachev/circle of chained words/Program.cs

```diff
@@ -0,0 +1,146 @@
+﻿/*
+Two words can be 'chained' if the last character of the first word is the same as the first character of the second word.
+
+Given a list of words, determine if there is a way to 'chain' all the words in a circle.
+
+Example:
+Input: ['eggs', 'karat', 'apple', 'snack', 'tuna']
+Output: True
+Explanation:
+The words in the order of ['apple', 'eggs', 'snack', 'karat', 'tuna'] creates a circle of chained words.
+*/
+
+
+/*
+
+Solution
+
+- can start from any word in the list, each word is a chance to build a full circle

```

Comment:
Huge thanks for explanation, I love the idea of having "used words" tracked. Makes total sense and I was think about it myself that I'm wasting time traversing through all words each time.

Reply:
You are welcome! I like to "explain" the code in advance in pseudo-code or otherwise, some advise to do so on live coding interviews, so that interviewer can undestand your way of thinking.

And on "used words", thanks for the support! 
I think this technique can be repeated in many other tasks where we can save the result in dictionary instead of traversing the list, and thus decrease the time complexity. E.g. O(N) instead of O(N*N)